test/jdk/java/awt/MenuItem/LotsOfMenuItemsTest.java

```diff
@@ -0,0 +1,109 @@
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Frame;
+import java.awt.Menu;
+import java.awt.MenuBar;
+import java.awt.MenuItem;
+import java.awt.event.ComponentAdapter;
+import java.awt.event.ComponentEvent;
+
+import java.util.List;
+
+/*
+ * @test
+ * @bug 4175790
+ * @requires os.family == "windows"
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @summary Win32: Running out of command ids for menu items
+ * @run main/manual LotsOfMenuItemsTest
+ */
+
+public class LotsOfMenuItemsTest extends ComponentAdapter {
+    private static final int NUM_WINDOWS = 400;
+    private static TestFrame firstFrame;
+    private static TestFrame testFrame;
+
+    public static void main(String[] args) throws Exception {
+        LotsOfMenuItemsTest obj = new LotsOfMenuItemsTest();
+        String INSTRUCTIONS = """
+                This test creates lots of frames with menu bars.
+                When it's done you will see two frames.
+                Try to select menu items from each of them.
+
+                If everything seems to work - test passed.
+                Click "Pass" button in the test harness window.
+
+                If test crashes on you - test failed.""";
+
+        PassFailJFrame.builder()
+                .title("LotsOfMenuItemsTest")
+                .instructions(INSTRUCTIONS)
+                .rows((int) INSTRUCTIONS.lines().count() + 2)
+                .columns(40)
+                .testUI(obj::createAndShowUI)
+                .build()
+                .awaitAndCheck();
+    }
+
+    private List<Frame> createAndShowUI() {
+        firstFrame = new TestFrame("First frame");
+        firstFrame.addComponentListener(this);
+
+        for (int i = 1; i < NUM_WINDOWS - 1; ++i) {
+            testFrame = new TestFrame("Running(" + i + ")...");
+            testFrame.setVisible(false);
+            testFrame.dispose();
+        }
+        testFrame = new TestFrame("Last Frame");
+        return List.of(firstFrame, testFrame);
+    }
+
+    @Override
+    public void componentShown(ComponentEvent e) {
+        PassFailJFrame.positionTestWindow(firstFrame,
+                PassFailJFrame.Position.HORIZONTAL);
+        testFrame.setLocation(firstFrame.getX(),
+                firstFrame.getY() + firstFrame.getHeight() + 8);
+    }
+
+    private static class TestFrame extends Frame {
+        static int n = 0;
+
+        public TestFrame(String title) {
+            super(title);
+            MenuBar mb = new MenuBar();
+            for (int i = 0; i < 10; ++i) {
+                Menu m = new Menu("Menu_" + (i + 1));
+                for (int j = 0; j < 20; ++j) {
+                    MenuItem mi = new MenuItem("Menu item " + ++n);
+                    m.add(mi);
+                }
+                mb.add(m);
+            }
+            setMenuBar(mb);
+            setSize(450, 150);
+        }
```

Comment:
The original test had `setVisible(true);` here. This is why the loop contains `setVisible(false)`.

Without showing the frames, the test may not reproduce the original problem reported in [JDK-4175790](https://bugs.openjdk.org/browse/JDK-4175790).

Reply:
I think the testFrame's are made invisble and disposed just after creating. Only the last frame and first frame is visible.
After the test update, **testFrame and firstFrame is setVisible by PassFailJFrame**.

Reply:
Updated the test to set the test frames visible, `setVisible(true)`;